### PR TITLE
[figgy] add figgy-web-staging1

### PR DESCRIPTION
**Pull request by kayiwa**
add figgy-web-staging1 vm

### CODE REVIEW

**Comment by kayiwa** [comment]
supercedes #3732

File: group_vars/figgy/staging.yml [hunk 1]
```diff
@@ -135,9 +135,10 @@ rails_app_vars:
   - name: PULFALIGHT_UNPUBLISHED_TOKEN
     value: '{{ vault_pulfalight_unpublished_auth_token }}'
 sidekiq_worker_name: figgy-workers
-passenger_ruby: "/usr/bin/ruby2.7"
-ruby_version_override: "ruby2.7"
-bundler_version: "2.3.18"
+passenger_ruby: "/usr/local/bin/ruby"
+install_ruby_from_source: true
+ruby_version_override: "ruby-3.1.0"
```
Comment (hackartisan) [inline group_vars/figgy/staging.yml:140]: figgy doesn't run on ruby 3 yet. it's a high priority but not finished. is it a blocker for this work?

Comment (acozine) [reply group_vars/figgy/staging.yml:140]: We should be able to install ruby 2.7.5 from source. +1 to separating the ruby upgrade from the OS upgrade.

**Comment by hackartisan** [comment]
Could some of this work have interfered with figgy-staging2? It was busted today and it turned out to be a failure to connect to its local redis. We changed the configuration in `/home/deploy/app_configs/figgy` to point to `localhost` instead of `figgy-staging2.princeton.edu` and that fixed it. But that configuration had been working fine for ages.

Also, there are some confusing IPs when you `ping figgy-staging2.princeton.edu` or run `nslookup figgy-staging2.princeton.edu` from the box itself.

```
# ping figgy-staging2.princeton.edu
PING figgy-staging2.princeton.edu (127.0.1.1) 56(84) bytes of data.
64 bytes from figgy-staging2.princeton.edu (127.0.1.1): icmp_seq=1 ttl=64 time=0.026 ms
```

```
# nslookup figgy-staging2.princeton.edu
Server:         127.0.0.53
Address:        127.0.0.53#53

Non-authoritative answer:
Name:   figgy-staging2.princeton.edu
Address: 127.0.1.1
```

We tell redis to bind to `127.0.0.1` -- if the local ip was redefined it maybe won't find redis anymore?

**Comment by acozine** [comment]
Current error is
```
"msg": "Error pulling image quay.io/pulibrary/jammy-ansible:latest - 500 Server Error for http+docker://localhost/v1.41/images/create?tag=latest&fromImage=quay.io%2Fpulibrary%2Fjammy-ansible: Internal Server Error (\"Head \"[https://quay.io/v2/pulibrary/jammy-ansible/manifests/latest\](https://quay.io/v2/pulibrary/jammy-ansible/manifests/latest/)": received unexpected HTTP status: 502 Bad Gateway\")"
```

**Comment by acozine** [comment]
This branch is now passing CI, but when I log into my test instance, I get the wrong version of Ruby:
```
root@instance:/# ruby -v
ruby 3.1.3p185 (2022-11-24 revision 1a6b16756e) [x86_64-linux]
```

**Review by acozine — CHANGES_REQUESTED** [review]
Currently still builds ruby 3.1.3. Don't merge until this is fixed!

**Comment by acozine** [comment]
Okay, on the test instance, this branch builds ruby 3.1.3, because the test environment is not pulling in group_vars. On a live server, this branch builds ruby 2.7.5. You must install openssl-1 as an option BEFORE you build Ruby, otherwise `gem install bundler` will fail with:
```
fatal: [figgy-web-staging-1.princeton.edu]: FAILED! => {"changed": true, "cmd": ["gem", "install", "bundler", "-v", "2.3.18"], "delta": "0:00:00.136166", "end": "2023-03-13 20:59:58.460344", "msg": "non-zero return code", "rc": 1, "start": "2023-03-13 20:59:58.324178", "stderr": "ERROR:  Loading command: install (LoadError)\n\tcannot load such file -- openssl\nERROR:  While executing gem ... (NoMethodError)\n    undefined method `invoke_with_build_args' for nil:NilClass", "stderr_lines": ["ERROR:  Loading command: install (LoadError)", "\tcannot load such file -- openssl", "ERROR:  While executing gem ... (NoMethodError)", "    undefined method `invoke_with_build_args' for nil:NilClass"], "stdout": "", "stdout_lines": []}
```
I installed openssl-1 manually:
```
wget https://www.openssl.org/source/openssl-1.1.1o.tar.gz
tar -xvzf openssl-1.1.1o.tar.gz
cd openssl-1.1.1o
./config && make && make test
export LD_LIBRARY_PATH=/opt/openssl-1/openssl-1.1.1o:$LD_LIBRARY_PAT
```
Then I deleted the old ruby installation and the next time I ran the playbook, it successfully installed the correct ruby version:
```
figgy-web-staging-1% ruby -v
ruby 2.7.5p203 (2021-11-24 revision f69aeb8314) [x86_64-linux]
```
The playbook also uninstalled some version of bundler and installed another version, but I don't think it's correct yet. Here's the output:
```
TASK [ruby_s : ruby_s | output existing bundler version] ***************************************************************************
ok: [figgy-web-staging-1.princeton.edu] => {
    "msg": "Installed bundler 2.1.4 Desired bundler 2.3.18"
}

TASK [ruby_s : ruby_s | install, upgrade, or downgrade bundler] ********************************************************************
included: /Users/alicia/git_repos/princeton_ansible/roles/ruby_s/tasks/install_bundler.yml for figgy-web-staging-1.princeton.edu

TASK [ruby_s : ruby_s | uninstall wrong version of bundler] ************************************************************************
changed: [figgy-web-staging-1.princeton.edu]

TASK [ruby_s : ruby_s | get gem home] **********************************************************************************************
changed: [figgy-web-staging-1.princeton.edu]

TASK [ruby_s : ruby_s | remove installed bundler if default] ***********************************************************************
changed: [figgy-web-staging-1.princeton.edu]

TASK [ruby_s : ruby_s | install global bundler, specific version] ******************************************************************
changed: [figgy-web-staging-1.princeton.edu]
```
However, when I SSH to the server, I see:
```
figgy-web-staging-1% bundler -v
Bundler version 2.4.8
```
I will start here tomorrow.

**Comment by kayiwa** [comment]
superceded by #3787

**Pull request closed without merging**